app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,46 @@
 class PostsController < ApplicationController
+  helper_method :current_user
+
   def index
-    @user = User.find(params[:id])
+    @user = User.find(params[:user_id])
     @posts = @user.posts.includes(comments: :author)
   end
 
   def show
     @post = Post.includes(comments: :author).find(params[:id])
-    @user = @post.author
+    @user = current_user
+    @new_like = Like.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post successfully created'
```

Comment:
:red_circle: When I try to create a new post line 24 throws the error below :arrow_down: 
![error](https://github.com/husush23/Blog-App/assets/26736582/5b0eccfc-1d97-4758-9db8-f9779454795f)

You are trying to access `@user`  which seems to be null. To fix this error make sure you set the @user every time you are creating a post.

Here is how to fix this: 

```
def create
    @post = current_user.posts.new(post_params)
    @user = User.find(params[:user_id])  #Add this line
    if @post.save
    #.... your code
end
```